test/test_repository.py

```diff
@@ -198,6 +198,35 @@ def test_hashfile(self):
         written_sha1 = self.repo.create_blob(data)
         self.assertEqual(hashed_sha1, written_sha1)
 
+    def test_conflicts_in_bare_repository(self):
+        def create_conflict_file(repo, branch, content):
+            oid = repo.create_blob(content)
+            tb = repo.TreeBuilder()
+            tb.insert('conflict', oid, pygit2.GIT_FILEMODE_BLOB)
+            tree = tb.write()
+
+            sig = pygit2.Signature('Author', 'author@example.com')
+            commit = repo.create_commit(branch.name, sig, sig,
+                    'Conflict', tree, [branch.target])
+            self.assertIsNotNone(commit)
+            return commit
+
+        b1 = self.repo.create_branch('b1', self.repo.head.peel())
+        c1 = create_conflict_file(self.repo, b1, 'Conflict 1')
+        b2 = self.repo.create_branch('b2', self.repo.head.peel())
+        c2 = create_conflict_file(self.repo, b2, 'Conflict 2')
+
+        index = self.repo.merge_commits(c1, c2)
+        self.assertIsNotNone(index.conflicts)
+
+        for (a, t, o) in index.conflicts:
```

Comment:
This loop would only ever loop once, and we know which file we want. Possibly asserting `len(index.conflicts)` and then looking up `(a, t, o) = index.conflicts["conflict"]` would show better that we're just testing the one file, rather than saying that every file conflict should be this way.